modules/core/keeper/keeper.go

```diff
@@ -86,7 +86,7 @@ func (k *Keeper) SetRouter(rtr *porttypes.Router) {
 
 // SetRouterV2 sets the v2 router for the IBC Keeper.
 func (k *Keeper) SetRouterV2(rtr *api.Router) {
-	k.PortKeeper.RouterV2 = rtr
+	k.ChannelKeeperV2.Router = rtr
```

Comment:
makes a bit more sense here I think, otherwise we introduce  a dependence of the port router on the ChannelKeeperV2